platforms/documentation/docs/src/snippets/bestPractices/avoidProviderGetOutsideTaskAction/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,32 @@
+// tag::provider-get-task-setup[]
+abstract class MyTask : DefaultTask() {
+    @get:Input
+    abstract val myInput: Property<String>
+
+    @get:OutputFile
+    abstract val myOutput: RegularFileProperty
+
+    @TaskAction
+    fun doAction() {
+        // Use get in the task action
+        val outputFile = myOutput.get().asFile
+        val outputText = myInput.get()
+        println(outputText)
+        outputFile.writeText(outputText)
+    }
+}
+// end::provider-get-task-setup[]
+
+// tag::avoid-this[]
+tasks.register<MyTask>("avoidThis") {
+    myInput = "currentEnvironment=${providers.gradleProperty("currentEnvironment").get()}"  // <1>
```

Comment:
💭 Using `providers.gradleProperty` is perhaps a little extra complexity than needed here to demo this, using another task with an output property would be the more common case - but would add length to the example.  

It might be slightly clearer to move this outside the task:

```val environment = providers.gradleProperty("currentEnvironment")```

to make it slightly more clear that the orign of the property is irrelevant here, the key part to focus on is the usage with `.get()`.

Reply:
Moved it to the set up part.